jetty-server/src/main/java/org/eclipse/jetty/server/session/AbstractSessionCache.java

```diff
@@ -343,113 +341,60 @@ public Session get(String id) throws Exception
-                                phsLock.close();
-                                break;
-                            }
-                            else
-                            {
-                                //successfully swapped in the session
-                                session.setResident(true);
-                                if (enter)
-                                    session.use();
-                                phsLock.close();
-                                break;
-                            }
-                        }
-                    }
-                    catch (Exception e)
-                    {
-                        ex = e; //remember a problem happened loading the session
-                        doDelete(id); //remove the placeholder
-                        phsLock.close();
-                        session = null;
-                        break;
-                    }
+                    return loadSession(k);
                 }
-                else
+                catch (Exception e)
                 {
-                    //my placeholder didn't win, check the session returned
-                    phsLock.close();
-                    try (Lock lock = s.lock())
-                    {
-                        //is it a placeholder? or is a non-resident session? In both cases, chuck it away and start again
-                        if (!s.isResident() || s instanceof PlaceHolderSession)
-                        {
-                            session = null;
-                            continue;
-                        }
-                        //I will use this session too
-                        session = s;
-                        if (enter)
-                            session.use();
-                        break;
-                    }
+                    LOG.warn("Error loading session {}", id, e);
+                    return null;
                 }
+            });
+
+            if (session == null) //session does not exist in store
+            {
+                if (LOG.isDebugEnabled())
+                    LOG.debug("Session {} not loaded by store", id);
+                return null;
             }
             else
             {
-                //check the session returned
                 try (Lock lock = session.lock())
                 {
-                    //is it a placeholder? or is it passivated? In both cases, chuck it away and start again
-                    if (!session.isResident() || session instanceof PlaceHolderSession)
-                    {
-                        session = null;
-                        continue;
-                    }
-
-                    //got the session
+                    session.setResident(true); //ensure freshly loaded session is resident
                     if (enter)
                         session.use();
-                    break;
+                }
+            }
+        }
+        else
+        {
```

Comment:
You wont need this else branch if you just go straight to computeIfAbsent above.

Reply:
Done.